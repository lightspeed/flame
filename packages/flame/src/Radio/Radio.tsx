import * as React from 'react';
import styled from '@emotion/styled';
import { themeGet } from '@styled-system/theme-get';
import { layout, LayoutProps } from 'styled-system';

import { Box } from '../Core';
import { BaseLabel, LabelProps, FormHelper } from '../FormField/FormField';

export interface RadioLabelProps extends LabelProps {}
const RadioLabel: React.FC<RadioLabelProps> = ({
  description,
  disabled,
  children,
  ...restProps
}) => {
  return (
    <Box>
      <BaseLabel {...restProps}>{children}</BaseLabel>
      {description && <FormHelper ml={['1.85rem', '1.75rem']}>{description}</FormHelper>}
    </Box>
  );
};

RadioLabel.defaultProps = {
  fontSize: ['text', 'text-s'],
  lineHeight: 3,
  fontWeight: 'bold',
};

const WrapperRadio = styled('div')`
  position: relative;
  line-height: 1rem;
`;

const Checkmark = styled('div')<LayoutProps>`
  position: relative;
  width: 1rem;
  height: 1rem;
  ${layout};
  flex: 0 0 auto;
  border-radius: ${themeGet('radii.radius-circle')};
  background-color: transparent;
  border: solid 1px ${themeGet('radioStyles.border')};
  box-sizing: border-box;
  overflow: hidden;
  background: ${themeGet('radioStyles.background')};
  margin-right: ${themeGet('space.2')};
`;

Checkmark.defaultProps = {
  width: ['18px', '16px'],
  height: ['18px', '16px'],
};

const Centermark = styled('div')<LayoutProps>`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  ${layout};
  border-radius: ${themeGet('radii.radius-circle')};
  background: ${themeGet('radioStyles.background')};
`;

Centermark.defaultProps = {
  width: ['7px', '6px'],
  height: ['7px', '6px'],
};

const RadioInput = styled('input')`
  position: absolute;
  ${layout};
  top: 0;
  left: 0;
  opacity: 0;
  padding: 0;
  margin: 0;
  z-index: 1;

  &:checked + ${Checkmark} {
    background: ${themeGet('radioStyles.checked.background')};
    border-color: ${themeGet('radioStyles.checked.border')};
  }

  &:not(:checked) + ${Checkmark} > ${Centermark} {
    display: none;
  }

  &:focus + ${Checkmark} {
    box-shadow: ${themeGet('radioStyles.focus.boxShadow')};
  }

  &[disabled] + ${Checkmark} {
    opacity: 0.5;
    box-shadow: none;
  }
`;

RadioInput.defaultProps = {
  // Fun clashing between legacy props and our custom props
  // @ts-ignore
  width: ['18px', '16px'],
  // @ts-ignore
  height: ['18px', '16px'],
};

export interface RadioProps extends React.InputHTMLAttributes<HTMLInputElement> {
  css?: any;
}

const BaseRadio = React.forwardRef<HTMLInputElement, RadioProps>(({ ...restProps }, ref) => (
  <WrapperRadio>
    <RadioInput ref={ref} type="radio" {...restProps} />
    <Checkmark>
      <Centermark data-testid="radio-centermark" />
    </Checkmark>
  </WrapperRadio>
));
interface FormRadioProps extends RadioProps {
  label?: React.ReactNode;
  description?: React.ReactNode;
  css?: any;
}
const Radio = React.forwardRef<HTMLInputElement, FormRadioProps>(
  ({ label, id, description, disabled, ...restProps }, ref) => {
    return (
      <React.Fragment>
        <RadioLabel htmlFor={id} disabled={disabled} description={description}>
          <BaseRadio ref={ref} id={id} disabled={disabled} {...restProps} />
          {label}
        </RadioLabel>
      </React.Fragment>
    );
  },
);

export { BaseRadio, RadioLabel, Radio };
