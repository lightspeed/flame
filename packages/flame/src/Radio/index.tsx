import { BaseRadio, Radio, RadioProps, RadioLabel, RadioLabelProps } from './Radio';

export { BaseRadio, Radio, RadioProps, RadioLabel, RadioLabelProps };
