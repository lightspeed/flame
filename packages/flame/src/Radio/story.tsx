import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withReadme } from 'storybook-readme';

import { Text, Heading2, Heading3, TextContent } from '../Text';
import RadioWrapper from './examples';
import { Radio, BaseRadio, RadioLabel } from './Radio';
import { Box } from '../Core';

import Readme from './README.md';

const stories = storiesOf('Components|Radio', module).addDecorator(withReadme(Readme));

stories.add('Story', () => (
  <div>
    <Heading3 mb={2}>Form Radio</Heading3>
    <Box mb={1}>
      <Radio name="animal_radio" id="radio_animal_1" label="Lion" />
    </Box>
    <Box mb={1}>
      <Radio name="animal_radio" id="radio_animal_2" label="Tiger" />
    </Box>
    <Box mb={1}>
      <Radio name="animal_radio" id="radio_animal_3" label="Bear" />
    </Box>
    <Box mb={1}>
      <Radio name="animal_radio" id="radio_animal_4" label="Lama" disabled />
    </Box>

    <Heading3 mb={2}>Radio with label and description</Heading3>
    <Box mb={3}>
      <Radio
        name="radiolabelanddescription"
        id="radiolabelanddescription"
        label="Label"
        description="Description text"
        checked={false}
      />
    </Box>

    <Heading3 mb={2}>Radio with long label and description in a 50% container</Heading3>
    <Box mb={3} style={{ width: '50%' }}>
      <Radio
        name="radiolong"
        id="radiolong"
        label="Long Label Long Label Long Label Long Label Long Label Long Label Long Label Long Label Long Label"
        description="Long Description Long Description Long Description Long Description Long Description Long Description Long Description Long Description"
        checked={false}
      />
    </Box>
    <Heading3 mb={2}>Radio with custom label and description</Heading3>
    <Box mb={3}>
      <Radio
        name="radio-custom"
        id="radio-custom"
        checked={false}
        label={<Text color="red">Customizing the radio label</Text>}
        description={
          <Text color="blue" fontSize="large" mt={2}>
            Customizing the radio label description
          </Text>
        }
      />
    </Box>

    <Heading3 mb={2}>Radio states</Heading3>
    <Box mb={3}>
      <Radio id="unchecked" label="Unchecked" checked={false} onChange={() => {}} />
    </Box>
    <Box mb={3}>
      <Radio id="checked" label="Checked" checked onChange={() => {}} />
    </Box>
    <Box mb={3}>
      <Radio id="disabled" label="Disabled" disabled checked={false} onChange={() => {}} />
    </Box>
    <Box mb={3}>
      <Radio id="checked-disabled" label="Checked Disabled" checked onChange={() => {}} disabled />
    </Box>
  </div>
));

stories.add('Rebuilding using the base components', () => (
  <div>
    <TextContent>
      <Heading2>Using BaseRadio</Heading2>
      <Heading3>Radio with label</Heading3>
      <Box mb={3}>
        <Text>To add a label and description, we can use the FormField component</Text>
        <RadioLabel htmlFor="radiolabel">
          <BaseRadio id="radiolabel" />
          Label
        </RadioLabel>
      </Box>
      <Box mb={3} style={{ width: '50%' }}>
        <RadioLabel
          htmlFor="radiolong"
          description="Long Description Long Description Long Description Long Description Long Description Long Description Long Description Long Description"
        >
          <BaseRadio id="radiolong" checked={false} onChange={() => {}} />
          Long Label Long Label Long Label Long Label Long Label Long Label Long Label Long Label
          Long Label
        </RadioLabel>
      </Box>
      <Heading3>Using custom elements</Heading3>
    </TextContent>
    <TextContent>
      <Heading2>Radio states</Heading2>
      <Box mb={3}>
        <RadioLabel htmlFor="unchecked">
          <BaseRadio id="unchecked" checked={false} />
          Unchecked
        </RadioLabel>
      </Box>
      <Box mb={3}>
        <RadioLabel htmlFor="checked">
          <BaseRadio id="checked" checked />
          Checked
        </RadioLabel>
      </Box>
      <Box mb={3}>
        <RadioLabel htmlFor="disabled">
          <BaseRadio id="disabled" disabled />
          <Text color="disabled">Disabled</Text>
        </RadioLabel>
      </Box>
      <Box mb={3}>
        <RadioLabel htmlFor="checked-disabled">
          <BaseRadio id="checked-disabled" disabled checked />
          <Text color="disabled">Checked Disabled</Text>
        </RadioLabel>
      </Box>
    </TextContent>
  </div>
));

stories.add(
  'Uncontrolled state',
  () => (
    <TextContent>
      <Heading2>Uncontrolled Radio states</Heading2>
      <Box mb={3}>
        <Radio id="option_1" name="test" label="Lion" />
      </Box>
      <Box mb={3}>
        <Radio id="option_2" name="test" label="Tiger" />
      </Box>
      <Box mb={3}>
        <Radio id="option_3" name="test" label="Bear" />
      </Box>
    </TextContent>
  ),
  { percy: { skip: true } },
);

stories.add(
  'Events',
  () => (
    <TextContent>
      <Heading2>Radio events</Heading2>
      <Box mb={3}>
        <Text as="div" fontWeight="bold" mb={1}>
          Input with onChange event
        </Text>
        <Radio
          id="onChange"
          label="Click me"
          onChange={action('onChange')}
          checked={false}
          className="cr-mb-1"
        />
        <Text as="div" size="small" color="gray-300">
          (See action logger)
        </Text>
      </Box>
      <Box mb={3}>
        <Text as="div" size="small" fontWeight="bold" mb={1}>
          Input with onFocus & onBlur events
        </Text>
        <Radio
          id="onFocusOnBlur"
          label="Click me"
          value="Test"
          onFocus={action('onFocus')}
          onBlur={action('onBlur')}
          onChange={() => {}}
          checked
          className="cr-mb-1"
        />
        <Text as="div" size="small" color="gray-300">
          (See action logger)
        </Text>
      </Box>
      <Box mb={3}>
        <Text as="div" size="small" fontWeight="bold" mb={1}>
          Input with state management
        </Text>
        <RadioWrapper action={action('onChange')} />
        <Text as="div" size="small" color="gray-300">
          (See action logger)
        </Text>
      </Box>
    </TextContent>
  ),
  { percy: { skip: true } },
);
