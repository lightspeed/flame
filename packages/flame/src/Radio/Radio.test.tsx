import * as React from 'react';
import { createComponent, customRender, fireEvent } from 'test-utils';
import { Radio } from './Radio';

describe('<Radio />', () => {
  describe('Snapshots', () => {
    it('renders a simple Radio correctly', () => {
      const component = createComponent(<Radio checked={false} onChange={jest.fn()} />);
      expect(component.toJSON()).toMatchSnapshot();
    });

    it('renders a Radio with a label and description correctly', () => {
      const component = createComponent(
        <Radio
          checked={false}
          onChange={jest.fn()}
          label="Label"
          description="Description"
          id="Radio"
          value="value"
        />,
      );
      expect(component.toJSON()).toMatchSnapshot();
    });

    it('renders a checked Radio correctly', () => {
      const component = createComponent(
        <Radio onChange={jest.fn()} id="Radio" value="value" checked />,
      );
      expect(component.toJSON()).toMatchSnapshot();
    });

    it('renders a disabled Radio correctly', () => {
      const component = createComponent(
        <Radio checked={false} onChange={jest.fn()} id="Radio" value="value" disabled />,
      );
      expect(component.toJSON()).toMatchSnapshot();
    });
  });

  describe('Events', () => {
    it('handles the `onChange` event', () => {
      const onChange = jest.fn();
      const { container } = customRender(<Radio checked={false} onChange={onChange} />);
      expect(onChange).not.toHaveBeenCalled();
      fireEvent.click(container.querySelector('input'));
      expect(onChange).toHaveBeenCalled();
    });

    it('handles the `onFocus` event', () => {
      const onFocus = jest.fn();
      const { container } = customRender(
        <Radio checked={false} onChange={jest.fn()} onFocus={onFocus} />,
      );
      expect(onFocus).not.toHaveBeenCalled();
      fireEvent.focus(container.querySelector('input'));
      expect(onFocus).toHaveBeenCalled();
    });

    it('handles the `onBlur` event', () => {
      const onBlur = jest.fn();
      const { container } = customRender(
        <Radio checked={false} onChange={jest.fn()} onBlur={onBlur} />,
      );
      expect(onBlur).not.toHaveBeenCalled();
      fireEvent.blur(container.querySelector('input'));
      expect(onBlur).toHaveBeenCalled();
    });
  });

  describe('Render', () => {
    const { getByText } = customRender(
      <Radio label={<div>my label</div>} description={<div>my description</div>} />,
    );

    getByText('my label');
    getByText('my description');
  });
});
