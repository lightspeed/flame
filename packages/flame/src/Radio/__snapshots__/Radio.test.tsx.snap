// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio /> Snapshots renders a Radio with a label and description correctly 1`] = `
.emotion-8 {
  margin: 0;
  font-size: 1rem;
  line-height: 1.125rem;
  font-weight: 700;
  color: #0c0d0d;
  font-weight: 700;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media screen and (min-width:601px) {
  .emotion-8 {
    font-size: 0.875rem;
  }
}

.emotion-6 {
  position: relative;
  line-height: 1rem;
}

.emotion-0 {
  position: absolute;
  width: 18px;
  height: 18px;
  top: 0;
  left: 0;
  opacity: 0;
  padding: 0;
  margin: 0;
  z-index: 1;
}

@media screen and (min-width:601px) {
  .emotion-0 {
    width: 16px;
    height: 16px;
  }
}

.emotion-0:checked + .emotion-5 {
  background: #2875c6;
  border-color: #1d5794;
}

.emotion-0:not(:checked) + .emotion-5 > .emotion-3 {
  display: none;
}

.emotion-0:focus + .emotion-5 {
  box-shadow: 0 0 0 1px #fff,0 0 0 3px #2875c6;
}

.emotion-0[disabled] + .emotion-5 {
  opacity: 0.5;
  box-shadow: none;
}

.emotion-4 {
  position: relative;
  width: 1rem;
  height: 1rem;
  width: 18px;
  height: 18px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  border-radius: 50%;
  background-color: transparent;
  border: solid 1px #abb3b3;
  box-sizing: border-box;
  overflow: hidden;
  background: #fff linear-gradient(180deg,#fff,#f3f3f3);
  margin-right: 0.75rem;
}

@media screen and (min-width:601px) {
  .emotion-4 {
    width: 16px;
    height: 16px;
  }
}

.emotion-2 {
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  width: 7px;
  height: 7px;
  border-radius: 50%;
  background: #fff linear-gradient(180deg,#fff,#f3f3f3);
}

@media screen and (min-width:601px) {
  .emotion-2 {
    width: 6px;
    height: 6px;
  }
}

.emotion-10 {
  margin: 0;
  margin-left: 1.85rem;
  font-size: 0.875rem;
  line-height: 1.125rem;
}

@media screen and (min-width:601px) {
  .emotion-10 {
    margin-left: 1.75rem;
  }
}

<div
  className="emotion-12 emotion-13"
>
  <label
    className="emotion-8 emotion-9"
    color="textHeading"
    fontSize={
      Array [
        "text",
        "text-s",
      ]
    }
    fontWeight="bold"
    htmlFor="Radio"
  >
    <div
      className="emotion-6 emotion-7"
    >
      <input
        checked={false}
        className="emotion-0 emotion-1"
        height={
          Array [
            "18px",
            "16px",
          ]
        }
        id="Radio"
        onChange={[MockFunction]}
        type="radio"
        value="value"
        width={
          Array [
            "18px",
            "16px",
          ]
        }
      />
      <div
        className="emotion-4 emotion-5"
        height={
          Array [
            "18px",
            "16px",
          ]
        }
        width={
          Array [
            "18px",
            "16px",
          ]
        }
      >
        <div
          className="emotion-2 emotion-3"
          data-testid="radio-centermark"
          height={
            Array [
              "7px",
              "6px",
            ]
          }
          width={
            Array [
              "7px",
              "6px",
            ]
          }
        />
      </div>
    </div>
    Label
  </label>
  <div
    className="emotion-10 emotion-11"
    size="small"
  >
    Description
  </div>
</div>
`;

exports[`<Radio /> Snapshots renders a checked Radio correctly 1`] = `
.emotion-8 {
  margin: 0;
  font-size: 1rem;
  line-height: 1.125rem;
  font-weight: 700;
  color: #0c0d0d;
  font-weight: 700;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media screen and (min-width:601px) {
  .emotion-8 {
    font-size: 0.875rem;
  }
}

.emotion-6 {
  position: relative;
  line-height: 1rem;
}

.emotion-0 {
  position: absolute;
  width: 18px;
  height: 18px;
  top: 0;
  left: 0;
  opacity: 0;
  padding: 0;
  margin: 0;
  z-index: 1;
}

@media screen and (min-width:601px) {
  .emotion-0 {
    width: 16px;
    height: 16px;
  }
}

.emotion-0:checked + .emotion-5 {
  background: #2875c6;
  border-color: #1d5794;
}

.emotion-0:not(:checked) + .emotion-5 > .emotion-3 {
  display: none;
}

.emotion-0:focus + .emotion-5 {
  box-shadow: 0 0 0 1px #fff,0 0 0 3px #2875c6;
}

.emotion-0[disabled] + .emotion-5 {
  opacity: 0.5;
  box-shadow: none;
}

.emotion-4 {
  position: relative;
  width: 1rem;
  height: 1rem;
  width: 18px;
  height: 18px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  border-radius: 50%;
  background-color: transparent;
  border: solid 1px #abb3b3;
  box-sizing: border-box;
  overflow: hidden;
  background: #fff linear-gradient(180deg,#fff,#f3f3f3);
  margin-right: 0.75rem;
}

@media screen and (min-width:601px) {
  .emotion-4 {
    width: 16px;
    height: 16px;
  }
}

.emotion-2 {
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  width: 7px;
  height: 7px;
  border-radius: 50%;
  background: #fff linear-gradient(180deg,#fff,#f3f3f3);
}

@media screen and (min-width:601px) {
  .emotion-2 {
    width: 6px;
    height: 6px;
  }
}

<div
  className="emotion-10 emotion-11"
>
  <label
    className="emotion-8 emotion-9"
    color="textHeading"
    fontSize={
      Array [
        "text",
        "text-s",
      ]
    }
    fontWeight="bold"
    htmlFor="Radio"
  >
    <div
      className="emotion-6 emotion-7"
    >
      <input
        checked={true}
        className="emotion-0 emotion-1"
        height={
          Array [
            "18px",
            "16px",
          ]
        }
        id="Radio"
        onChange={[MockFunction]}
        type="radio"
        value="value"
        width={
          Array [
            "18px",
            "16px",
          ]
        }
      />
      <div
        className="emotion-4 emotion-5"
        height={
          Array [
            "18px",
            "16px",
          ]
        }
        width={
          Array [
            "18px",
            "16px",
          ]
        }
      >
        <div
          className="emotion-2 emotion-3"
          data-testid="radio-centermark"
          height={
            Array [
              "7px",
              "6px",
            ]
          }
          width={
            Array [
              "7px",
              "6px",
            ]
          }
        />
      </div>
    </div>
  </label>
</div>
`;

exports[`<Radio /> Snapshots renders a disabled Radio correctly 1`] = `
.emotion-8 {
  margin: 0;
  font-size: 1rem;
  line-height: 1.125rem;
  font-weight: 700;
  color: #0c0d0d;
  font-weight: 700;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media screen and (min-width:601px) {
  .emotion-8 {
    font-size: 0.875rem;
  }
}

.emotion-6 {
  position: relative;
  line-height: 1rem;
}

.emotion-0 {
  position: absolute;
  width: 18px;
  height: 18px;
  top: 0;
  left: 0;
  opacity: 0;
  padding: 0;
  margin: 0;
  z-index: 1;
}

@media screen and (min-width:601px) {
  .emotion-0 {
    width: 16px;
    height: 16px;
  }
}

.emotion-0:checked + .emotion-5 {
  background: #2875c6;
  border-color: #1d5794;
}

.emotion-0:not(:checked) + .emotion-5 > .emotion-3 {
  display: none;
}

.emotion-0:focus + .emotion-5 {
  box-shadow: 0 0 0 1px #fff,0 0 0 3px #2875c6;
}

.emotion-0[disabled] + .emotion-5 {
  opacity: 0.5;
  box-shadow: none;
}

.emotion-4 {
  position: relative;
  width: 1rem;
  height: 1rem;
  width: 18px;
  height: 18px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  border-radius: 50%;
  background-color: transparent;
  border: solid 1px #abb3b3;
  box-sizing: border-box;
  overflow: hidden;
  background: #fff linear-gradient(180deg,#fff,#f3f3f3);
  margin-right: 0.75rem;
}

@media screen and (min-width:601px) {
  .emotion-4 {
    width: 16px;
    height: 16px;
  }
}

.emotion-2 {
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  width: 7px;
  height: 7px;
  border-radius: 50%;
  background: #fff linear-gradient(180deg,#fff,#f3f3f3);
}

@media screen and (min-width:601px) {
  .emotion-2 {
    width: 6px;
    height: 6px;
  }
}

<div
  className="emotion-10 emotion-11"
>
  <label
    className="emotion-8 emotion-9"
    color="textHeading"
    fontSize={
      Array [
        "text",
        "text-s",
      ]
    }
    fontWeight="bold"
    htmlFor="Radio"
  >
    <div
      className="emotion-6 emotion-7"
    >
      <input
        checked={false}
        className="emotion-0 emotion-1"
        disabled={true}
        height={
          Array [
            "18px",
            "16px",
          ]
        }
        id="Radio"
        onChange={[MockFunction]}
        type="radio"
        value="value"
        width={
          Array [
            "18px",
            "16px",
          ]
        }
      />
      <div
        className="emotion-4 emotion-5"
        height={
          Array [
            "18px",
            "16px",
          ]
        }
        width={
          Array [
            "18px",
            "16px",
          ]
        }
      >
        <div
          className="emotion-2 emotion-3"
          data-testid="radio-centermark"
          height={
            Array [
              "7px",
              "6px",
            ]
          }
          width={
            Array [
              "7px",
              "6px",
            ]
          }
        />
      </div>
    </div>
  </label>
</div>
`;

exports[`<Radio /> Snapshots renders a simple Radio correctly 1`] = `
.emotion-8 {
  margin: 0;
  font-size: 1rem;
  line-height: 1.125rem;
  font-weight: 700;
  color: #0c0d0d;
  font-weight: 700;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media screen and (min-width:601px) {
  .emotion-8 {
    font-size: 0.875rem;
  }
}

.emotion-6 {
  position: relative;
  line-height: 1rem;
}

.emotion-0 {
  position: absolute;
  width: 18px;
  height: 18px;
  top: 0;
  left: 0;
  opacity: 0;
  padding: 0;
  margin: 0;
  z-index: 1;
}

@media screen and (min-width:601px) {
  .emotion-0 {
    width: 16px;
    height: 16px;
  }
}

.emotion-0:checked + .emotion-5 {
  background: #2875c6;
  border-color: #1d5794;
}

.emotion-0:not(:checked) + .emotion-5 > .emotion-3 {
  display: none;
}

.emotion-0:focus + .emotion-5 {
  box-shadow: 0 0 0 1px #fff,0 0 0 3px #2875c6;
}

.emotion-0[disabled] + .emotion-5 {
  opacity: 0.5;
  box-shadow: none;
}

.emotion-4 {
  position: relative;
  width: 1rem;
  height: 1rem;
  width: 18px;
  height: 18px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  border-radius: 50%;
  background-color: transparent;
  border: solid 1px #abb3b3;
  box-sizing: border-box;
  overflow: hidden;
  background: #fff linear-gradient(180deg,#fff,#f3f3f3);
  margin-right: 0.75rem;
}

@media screen and (min-width:601px) {
  .emotion-4 {
    width: 16px;
    height: 16px;
  }
}

.emotion-2 {
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  width: 7px;
  height: 7px;
  border-radius: 50%;
  background: #fff linear-gradient(180deg,#fff,#f3f3f3);
}

@media screen and (min-width:601px) {
  .emotion-2 {
    width: 6px;
    height: 6px;
  }
}

<div
  className="emotion-10 emotion-11"
>
  <label
    className="emotion-8 emotion-9"
    color="textHeading"
    fontSize={
      Array [
        "text",
        "text-s",
      ]
    }
    fontWeight="bold"
  >
    <div
      className="emotion-6 emotion-7"
    >
      <input
        checked={false}
        className="emotion-0 emotion-1"
        height={
          Array [
            "18px",
            "16px",
          ]
        }
        onChange={[MockFunction]}
        type="radio"
        width={
          Array [
            "18px",
            "16px",
          ]
        }
      />
      <div
        className="emotion-4 emotion-5"
        height={
          Array [
            "18px",
            "16px",
          ]
        }
        width={
          Array [
            "18px",
            "16px",
          ]
        }
      >
        <div
          className="emotion-2 emotion-3"
          data-testid="radio-centermark"
          height={
            Array [
              "7px",
              "6px",
            ]
          }
          width={
            Array [
              "7px",
              "6px",
            ]
          }
        />
      </div>
    </div>
  </label>
</div>
`;
